odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,254 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+"""Single-photon emission computerized tomography (SPECT) projectors """
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+try:
+    from NiftyPy.NiftyRec import (SPECT_project_parallelholes,
+                                  SPECT_backproject_parallelholes)
+    NIFTYREC_AVAILABLE = True
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+
+from odl.operator import Operator
+from odl import uniform_discr_frompartition
+
+__all__ = ('SpectProject', 'SpectBackproject')
+_SUPPORTED_IMPL = ('niftyrec', 'niftyrec_gpu')
+
+
+class SpectProject(Operator):
+    """Discrete SPECT projector"""
+    def __init__(self, dom, geometry, attenuation=None, psf=None,
+                 impl='niftyrec'):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, a 3 dimensional domain of the forward projector
+        geometry : geometry from :class:`~odl.tomo.geometry.spect`
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator domain
+        attenuation : `element-like` structure of the shape
+            `dom.shape`, optional
+            Linear attenuation map for SPECT.
+        psf : `element-like` structure, optional
+            Point Spread Functions for the correction of the
+            collimator-detector response
+        impl={'niftyrec', 'niftyrec_gpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+
+            'niftyrec' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `niftypy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and CPU
+
+            'niftyrec_gpu' : `NiftyRec <http://niftyrec.scienceontheweb.net/\
+wordpress/>`_ tomography toolbox using the `NiftyPy <https://github.com/\
+spedemon/niftypy>`_ Python wrapper and GPU
```

Comment:
This looks a bit ugly in the parameter list. I suggest making a `References` section by adding
```
References
----------
.. _NiftyRec: http://niftyrec.scienceontheweb.net/wordpress/
.. _NiftyPy: https://github.com/spedemon/niftypy
```
and referring in the text by
```
`NiftyRec`_
```
and
```
`NiftyPy`_
```
respectively.